### ISSUE TITLE
[Bug] [RLlib] Normalized actions wrongly clipped with GaussianNoise exploration 

### ISSUE THREAD

**Issue by gabrielmittag** [ray-project/ray#22835]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

RLlib

### What happened + What you expected to happen

The GaussianNoise exploration clips the gaussian sampled actions to the action space bounds. However, at this stage, the actions are already normalized. For example, let's assume an action space of [0,1], then the action of 0 will be normalized to -1. The action=-1 will then wrongly be cut off to 0 inside the GaussianNoise object since that is the lower action range.

https://github.com/ray-project/ray/blob/3d633132652b77eba2ceeb99614d96536440e9ea/rllib/utils/exploration/gaussian_noise.py#L200-L211

### Versions / Dependencies

Ray version 1.10.0

### Reproduction script

```
from ray.rllib.agents.ppo import PPOTrainer
from ray.rllib.models.torch.torch_modelv2 import TorchModelV2
import gym
import torch
import numpy as np

class Env(gym.Env):
    def __init__(self, args):
            self.action_space = gym.spaces.Box(low=0., high=1., shape=(1,), dtype=np.float64)
            self.observation_space = gym.spaces.Box(low=0., high=1., shape=(1,), dtype=np.float64)
    def step(self, action):
            print(f"action: {action[0]}")
            obs = action
            reward = 1.
            done = True
            info = {}
            return obs, reward, done, info
    def reset(self):
            state = np.array([1.])
            return state

class Net(TorchModelV2, torch.nn.Module):
    def __init__(self, obs_space, action_space, num_outputs, model_config,
                 name):
        TorchModelV2.__init__(self, obs_space, action_space, num_outputs,
                              model_config, name)
        torch.nn.Module.__init__(self)
        self.lin = torch.nn.Linear(1, 1)
    def forward(self, input_dict, state, seq_lens):
        obs = input_dict["obs"].float()
        action = torch.ones_like(obs) * (-1) # set to fixed normalized action 0.0 -> -1.0
        out = torch.cat((action, action), 1)
        return out, []
    def value_function(self):
        return torch.tensor([0.], requires_grad=True)

config = {
    "env": Env,
    "num_workers": 0,
    "train_batch_size": 5,
    "sgd_minibatch_size": 5,
    "rollout_fragment_length": 5,
    "framework": "torch",
    "model": {
        "custom_model": Net,
    },
    "normalize_actions": True,
    "explore": True,
    "exploration_config": {
        "type": "GaussianNoise",
        "stddev": 0.0,
        "random_timesteps": 0,
    }
}

# run with exploration but stdev set to 0.0 - actions will be wrongly cut off
trainer = PPOTrainer(config=config)
results = trainer.train()

# run without exploration - action won't be cut off
config["explore"] = False
trainer = PPOTrainer(config=config)
results = trainer.train()

```

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by sven1977** [comment]
Hey @gabrielmittag , thanks for filing this issue. Great catch! Would you be willing to provide a PR to fix this? You have already laid out the solution here, so you would be the best candidate for this :)
